src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -38,41 +47,26 @@ private enum MessageType
         Error
     }
 
-    private class Message
-    {
-        public Message(MessageType messageType, string finalMessage)
-        {
-            MessageType = messageType;
-            FinalMessage = finalMessage;
-        }
-
-        public MessageType MessageType { get; }
-        public string FinalMessage {  get; }
-    }
+    /// <summary>
+    /// List of UI warnings that should be logged.
+    /// </summary>
+    private readonly IList<string> uiWarnings = [];
 
-    public const ConsoleColor DebugColor = ConsoleColor.DarkCyan;
-    public const ConsoleColor WarningColor = ConsoleColor.Yellow;
-    public const ConsoleColor ErrorColor = ConsoleColor.Red;
+    private readonly IOutputWriter outputWriter;
 
-    private const LoggerVerbosity DefaultVerbosity = VerbosityCalculator.InitialLoggingVerbosity;
+    private readonly IFileWrapper fileWrapper = FileWrapper.Instance;
```

Comment:
Please use dependency injection for this one.
That's why we have the file wrapper, so you can inject the `IFileWrapper` interface.

Reply:
And once we have this, we should be able to write the ConsoleLogger test without touching the disk.